internal/kubernetes/networkHandler.go

```diff
@@ -87,6 +89,44 @@ func (n *networkHandler) OnHandshakeSuccess(username string, clientVersion strin
 		if n.pod, err = n.cli.createPod(ctx, n.labels, n.annotations, env, nil, nil); err != nil {
 			return nil, err
 		}
+		if b64creds, ok := metadata["KRB5CCDATA"]; ok {
```

Comment:
The Kubernetes backend shouldn't know anything about Kerberos. This way we'll have to implement Kerberos handling in all backends separately. Furthermore, the metadata can be exposed to the container as env variables anyway. I would suggest adding this logic to the agent as that will result in a much cleaner implementation. That would make it possible to pass this through via the SSH proxy too, not just for container backends.

If that doesn't work we need to make an API for deploying files, which I'm not a fan of because it will introduce added complexity. Furthermore, it will be difficult to make sure that in the future this doesn't escalate into all kinds of shenanigans.